src/nl/rubensten/texifyidea/intentions/latexmathtoggle/MathEnvironmentEditor.kt

```diff
@@ -0,0 +1,119 @@
+package nl.rubensten.texifyidea.intentions.latexmathtoggle
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import nl.rubensten.texifyidea.util.*
+
+class MathEnvironmentEditor(private val oldEnvName: String,
+                            private val newEnvName: String,
+                            val editor: Editor,
+                            val environment: PsiElement) {
+    /**
+     * Apply the conversion of a math environment.
+     */
+    fun apply() {
+        val document = editor.document
+        val indent = document.lineIndentationByOffset(environment.textOffset)
+
+        // The number of characters to replace before the begin block of the old environment starts.
+        val whitespace = when {
+            newEnvName == "inline" -> indent.length + 1
+            oldEnvName == "inline" -> 1
+            else -> 0
+        }
```

Comment:
While `when` is not wrong here, I'd prefer an `if`/`else` as it makes more sense. You're not pattern matching.